modiscolite/meme_writer.py

```diff
@@ -0,0 +1,158 @@
+# Create a class that can be used to write MEME files.
+# Authors: Ivy Raine <ivy.ember.raine@gmail.com>, Jacob Schreiber <jmschreiber91@gmail.com>
+from typing import List, Optional
+import numpy as np
+
+from os import PathLike
+
+class MEMEWriterMotif:
+	"""Class for handling Motif for MEME file writing."""
+
+	def __init__(
+		self,
+		name: str,
+		probability_matrix: np.ndarray,
+		# `alphabet_length` is known as 'alength' in MEME Suite.
+		alphabet_length: int,
+		# `source_sites` is known as 'nsites' in MEME Suite.
+		source_sites: int,
+		e_value: Optional[str] = None,
+		url: Optional[str] = None
+	) -> None:
+		self._name = name
```

Comment:
Why are you putting an underscore before the variable names and then separately have property functions returning those variable names? Wouldn't it be easier to just to `self.name = name` and then they can access `self.name`?

Reply:
I'm creating a getter/setter here because of potential side effects that we as the developer may want to implement down the line if the user accesses these properties. I agree it may feel redundant. I can remove it if you'd like, but I don't think it's a big issue. It's nice to have if we end up developing the meme writer script further and end up wanting to implement a side effect when accessing one of these properties. I usually avoid premature optimizations, but this is more of something that'll help us avoid refactoring later down the line for minimal cost.

Reply:
Removed them for now